src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/ContactsController.php

```diff
@@ -0,0 +1,192 @@
+use PrestaShopBundle\Security\Annotation\AdminSecurity;
+use PrestaShopBundle\Security\Annotation\DemoRestricted;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Template;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * ContactsController is responsible for actions and rendering
+ * of "Shop Parameters > Contact > Contacts" page.
+ */
+class ContactsController extends FrameworkBundleAdminController
+{
+    /**
+     * Shows page content.
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))")
+     *
+     * @Template("@PrestaShop/Admin/Configure/ShopParameters/Contact/Contacts/contacts.html.twig")
+     *
+     * @param Request $request
+     * @param ContactFilters $filters
+     *
+     * @return array
+     */
+    public function indexAction(Request $request, ContactFilters $filters)
+    {
+        $contactsGridFactory = $this->get('prestashop.core.grid.factory.contacts');
+        $gridPresenter = $this->get('prestashop.core.grid.presenter.grid_presenter');
+
+        $contactsGrid = $contactsGridFactory->getGrid($filters);
+
+        return [
+            'layoutTitle' => $this->trans('Contacts', 'Admin.Navigation.Menu'),
+            'layoutHeaderToolbarBtn' => [
+                'add' => [
+                    'href' => $this->generateUrl('admin_contacts_create'),
+                    'desc' => $this->trans('Add new contact', 'Admin.Shopparameters.Feature'),
+                    'icon' => 'add_circle_outline',
+                ],
+            ],
+            'requireAddonsSearch' => true,
+            'enableSidebar' => true,
+            'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+            'grid' => $gridPresenter->present($contactsGrid),
+        ];
+    }
+
+    /**
+     * Grid search action.
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function searchAction(Request $request)
+    {
+        $gridDefinitionFactory = $this->get('prestashop.core.grid.definition.factory.contacts');
+        $contactsGridDefinition = $gridDefinitionFactory->getDefinition();
+
+        $gridFilterFormFactory = $this->get('prestashop.core.grid.filter.form_factory');
+        $filtersForm = $gridFilterFormFactory->create($contactsGridDefinition);
+        $filtersForm->handleRequest($request);
+
+        $filters = [];
+
+        if ($filtersForm->isSubmitted()) {
+            $filters = $filtersForm->getData();
+        }
+
+        return $this->redirectToRoute('admin_contacts', ['filters' => $filters]);
+    }
+
+    /**
+     * Display the Contact creation form.
+     *
+     * @AdminSecurity("is_granted('create', request.get('_legacy_controller'))", message="You do not have permission to add this.")
```

Comment:
if user has access to: create, update, delete. Does it provide access to the page? same for other actions.

Reply:
If you don't have `read` access you cannot access the contact list page, as well as the contact creation form. Should it be accessible with such permissions?

Reply:
I'm sorry I dont understand the issue here